### ISSUE TITLE
Style/NegatedIfElseCondition: False positive on `if !!`

### ISSUE THREAD

**Issue by pdobb** [rubocop-hq/rubocop#9050]
Using `!!` isn't negation, it's conversion to Boolean. So this should be valid code:

```ruby
if !!my_method_or_variable
  "true condition"
else
  "false condition"
end

# Fails with:
Style/NegatedIfElseCondition: Invert the negated condition and swap the if-else branches.
  if !!my_method ...
  ^^^^^^^^^^^^^^
```

--------

## Expected behavior

Don't fail when `!!` is used in an if/else block. i.e. this should be valid:

```ruby
if !!my_method_or_variable
  "true condition"
else
  "false condition"
end
```

## Actual behavior

The Style/NegatedIfElseCondition cop fails with:
> Style/NegatedIfElseCondition: Invert the negated condition and swap the if-else branches

## Steps to reproduce the problem

Add an if/else such as the following to a ruby file and run the Style/NegatedIfElseCondition cop:

```ruby
if !!my_method
  "a"
else
  "b"
end
```

## RuboCop version

```
$ rubocop -V
1.3.0 (using Parser 2.7.2.0, rubocop-ast 1.1.1, running on ruby 2.6.6 x86_64-darwin19)
  - rubocop-performance 1.8.1
  - rubocop-rails 2.8.1
  - rubocop-rake 0.5.1
```
